linera-client/src/storage.rs

```diff
@@ -335,41 +343,59 @@ impl StorageConfigNamespace {
             #[cfg(feature = "storage-service")]
             StorageConfig::Service { endpoint } => {
                 let endpoint = endpoint.clone();
-                let config = ServiceStoreConfig {
+                let inner_config = ServiceStoreInternalConfig {
                     endpoint,
-                    common_config,
+                    common_config: common_config.reduced(),
+                };
+                let config = ServiceStoreConfig {
+                    inner_config,
+                    cache_size: common_config.cache_size,
                 };
                 Ok(StoreConfig::Service(config, namespace))
             }
             StorageConfig::Memory => {
-                let config = MemoryStoreConfig { common_config };
+                let config = MemoryStoreConfig {
+                    common_config: common_config.reduced(),
+                };
                 Ok(StoreConfig::Memory(config, namespace))
             }
             #[cfg(feature = "rocksdb")]
             StorageConfig::RocksDb { path, spawn_mode } => {
                 let path_buf = path.to_path_buf();
                 let path_with_guard = PathWithGuard::new(path_buf);
-                let config = RocksDbStoreConfig {
+                let inner_config = RocksDbStoreInternalConfig {
                     path_with_guard,
                     spawn_mode: *spawn_mode,
-                    common_config,
+                    common_config: common_config.reduced(),
+                };
+                let config = RocksDbStoreConfig {
+                    inner_config,
+                    cache_size: common_config.cache_size,
                 };
                 Ok(StoreConfig::RocksDb(config, namespace))
             }
             #[cfg(feature = "dynamodb")]
             StorageConfig::DynamoDb { use_localstack } => {
                 let aws_config = get_config(*use_localstack).await?;
-                let config = DynamoDbStoreConfig {
+                let inner_config = DynamoDbStoreInternalConfig {
                     config: aws_config,
-                    common_config,
+                    common_config: common_config.reduced(),
+                };
+                let config = DynamoDbStoreConfig {
+                    inner_config,
+                    cache_size: common_config.cache_size,
```

Comment:
The `cache_size` belongs to a `LruCacheConfig`.

Reply:
Can we replace `DynamoDbStoreConfig` by `LruCacheConfig<DynamoDbStoreConfig>` and rename `DynamoDbStoreInternalConfig` into `DynamoDbStoreConfig` ?

Reply:
I understand. Indeed this PR does not implement that idea. However, that can be done next.

Thus the next step would be to remove the "Internal" attribute to the `RockDb`, `ScyllaDb`, `DynamoDb` and others. And they could then be exposed to users so that users could run, e.g. with `ScyllaDb` with cache or without.

But I think this can be done in a subsequent work.